### PR TITLE
Api activities

### PR DESCRIPTION
**Pull request by gerhard6380**
replace return of api from number of activities (like/dislike/attendyes/attendno/attendmaybe) to arrays of the users for each activity. Return for each user has now the data structure as user fields in other API calls. Output works for JSON as expected. It also returns correct data for XML, but this was not tested so far with any client.

### CODE REVIEW

File: include/api.php [hunk 1]
```diff
@@ -2362,29 +2362,54 @@ function api_format_items_activities(&$item, $type = "json") {
...
 			$xml_activities = array();
-			foreach ($activities as $k => $v)
+			foreach ($activities as $k => $v) {
+				// change xml element from "like" to "friendica:like"
 				$xml_activities["friendica:".$k] = $v;
-
+				// add user data into xml output
+				$k_user = 0;
+				foreach ($v as $user)
+					$xml_activities["friendica:".$k][$k_user++.":user"] = $user;
```
Comment (fabrixxm) [inline include/api.php]: I'm not sure about `[$k_user++.":user"]` as, if I remember correctly, is "namespace:tagname", so you'll get invalid xml
(I've not tried the code yet)

**Comment by fabrixxm** [comment]
And I think there are commit in this PR that aren't related to the topic...

File: include/api.php (hunk 1 above)
Comment (gerhard6380) [inline include/api.php]: yeah, this is the magic line which made it working. How does it works? I really don't know! One would expect <k_user:user> but it returns <user>. The XML output now looks like this 
`<friendica:activities>
   <friendica:like>
      <user>
         <id>x</id>
         ...
         <network>dfrn</network>
      </user>
      <user>
         <id>x</id>
         ...
         <network>dfrn</network>
      </user>
   </friendica:like>
   <friendica:dislike>
      <user> ... </user>
   </friendica:dislike>
   <friendica:attendyes/>
   <friendica:attendno/>
   <friendica:attendmaybe/>
</friendica:activities>`
This seems to me as a correct xml syntax. \<user\> should be ok for me with my limited knowledge about xml, as it contains the same information as the \<user\> field has in Twitter API (no namespace => xmlns="http://api.twitter.com"). But if you prefer, we can change this to \<friendica:user\>. It doesn't matter for me as I am using json only for the Windows10 client.

**Comment by gerhard6380** [comment]
Yes, sorry. I came from a branch created with Github Desktop from develop. But @tobiasd told me to change the PR from develop to 3.5rc branch on friendica/friendica. After I did this I have now some more files in the PR. I should do a rebase, but I need to find out how this works in Github Desktop.

File: include/api.php (hunk 1 above)
Comment (fabrixxm) [inline include/api.php]: > The XML output now looks like this  [...]

that's perfect. I was commenting without trying the code, and I was wrong :-)

> we can change this to &lt;friendica:user&gt;

If it has the same structure as &lt;user&gt; elsewhere, I'll leave as it is. It's the same data, the same structure, the same tag...

**Pull request closed without merging**